hw/ip/sram_ctrl/doc/_index.md

```diff
@@ -19,12 +19,33 @@ The SRAM controller contains the SRAM data and address scrambling device and pro
 
 # Theory of Operations
 
-### Block Diagram
+## Block Diagram
 
-**TODO: draw block diagram and add description**
+As shown in the blockdiagram below (for `Width = 32`), the SRAM controller contains a CSR node, a key request interface, a TL-UL SRAM adapter and an instance of `prim_ram_1p_scr` that implements the actual scrambling algorithm.
+Scrambling is always enabled, but the scrambling device uses an all-zero scrambling key and nonce when it comes out of reset.
+It is the task of SW to request a new scrambling key and nonce via the CSRs as described in the [Programmer's Guide]({{< relref "#programmers-guide" >}}) below.
 
 ![SRAM Controller Block Diagram](sram_ctrl_blockdiag.svg)
 
+The scrambling device employs a reduced-round (5 instead of 11) PRINCE block cipher in CTR mode to scramble the data.
+The PRINCE lightweight block cipher has been selected due to its low latency and low area characteristics, see also [prim_prince]({{< relref "hw/ip/prim/doc/prim_prince" >}}) for more information on PRINCE.
+The number of rounds is reduced to 5 in order to ease timing pressure and ensure single cycle operation (the number of rounds can always be increased if it turns out that there is enough timing slack).
+
+In [CTR mode](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Counter_(CTR)), the block cipher is used to encrypt a 64bit IV with the scrambling key in order to create a 64bit keystream block that is bitwise XOR'ed with the data in order to transform plaintext into ciphertext and vice versa.
+The IV is assembled by concatenating a nonce with the word address.
+
+If the input data word is smaller than 64bit, the keystream block is truncated to fit the input data width.
+If the input data word is wider than 64bit, the SRAM controller by default instantiates multiple PRINCE primitives in order to create a unique keystream for the full datawidth.
+For area constrained settings, the parameter `ReplicateKeyStream` can be set to 1 in order to replicate the keystream block generated by one single primitive instead of using multiple parallel PRINCE instances (but it should be understood that this lowers the level of security).
+
+Since plain CTR mode does not diffuse the data bits due to the bitwise XOR, the scheme is augmented by passing each individual byte through a two-layer substitution-permutation (S&P) network implemented with the `prim_subst_perm` primitive.
```

Comment:
just to make sure i understand correctly, nothing happens to the data in between getting output from PRINCE and getting input to the custom S&P network?

Reply:
So actually, the data itself is not passed through PRINCE. PRINCE is used to generate a keystream that is XOR'ed over the data (see block diagram).

However, you understood correctly that after the keystream XOR, the data is directly fed into the custom S&P network and then into the memory.

Reply:
ahh sorry that is what I meant.
got it, thanks!
LGTM